### DISCUSSION TITLE
Email Template and redirects

### DISCUSSION THREAD

**Discussion by chmcquee** [keycloakify/keycloakify#251] [Q&A]
Hi all,
I'm continuing the journey of implementing keycloaks forgot password feature with keycloakify.
Sofar I have a question as to how to trigger sending out the email as verification after clicking the forgot password button in the Login screen.
I then also want to redirect to the login-verify-email.ftl page once the email has been sent out.
Then the link sent to the user should open the forgot password page (login-update-password.ftl) followed by another confirmation page.
I didn't find a way to link these pages without loosing the form submission. The default for all these forms is a redirect to the login screen which I only want in the end.

Appreciate any help you can offer 👍

**Comment by chmcquee** [comment]
@garronej is there a place where you explain how to integrate email-service into the app? I wanted to test it with onyxia but don't have a french email account

**Comment by garronej** [comment]
Hi  @chmcquee,  

Here you have a step by step documentation of how we setup Keycloak in Onyxia: https://docs.onyxia.sh/#enabling-user-authentication  
You can create a mail server with AWS SES.  

With this informations you should be able to make it work. :)  

PS: Theses concern are out of scrope as far as Keycloakify is concerned.

**Reply by chmcquee** [reply]
Thanks for the link, I'll look into that for the email

**Comment by chmcquee** [comment]
What I'm still stuck on is a redirect to other pages when successful. So I want to ask a user for their email when they forget their password. This is the "login-reset-password.ftl" file. I then want a custom page that shows the user that an email has been sent, instead of the success message from keycloak.
To do this I edited the "login-verify-email.ftl" page to show the instructions given on the screen. 
My question is how I can link these two pages. I tried react router and failed on the kcContext part.
The kcContext has a nested URL array that only has redirects for the loginAction and login. 
Thanks again!
